core/src/main/scala/org/apache/spark/status/api/v1/api.scala

```diff
@@ -86,7 +86,7 @@ class ExecutorStageSummary private[spark](
     val isBlacklistedForStage: Boolean,
     @JsonSerialize(using = classOf[ExecutorMetricsJsonSerializer])
     @JsonDeserialize(using = classOf[ExecutorMetricsJsonDeserializer])
-    val peakMemoryMetrics: Option[ExecutorMetrics],
+    val peakMemoryMetrics: ExecutorMetrics,
```

Comment:
How about checking whether `row` has `peakMemoryMetrics` in `stagepages.js`?

Reply:
> How about checking whether `row` has `peakMemoryMetrics` in `stagepages.js` rather than changing the API?

Updated.